bindata/network/ovn-kubernetes/ovnkube-master.yaml

```diff
@@ -145,7 +155,9 @@ spec:
               - -c
               - |
                 retries=0
-                while ! ovn-sbctl -t 5 set-connection ptcp:{{.OVN_SB_PORT}} -- set connection . inactivity_probe=0; do
+                while ! ( ovn-sbctl set-ssl /ovn-cert/tls.key /ovn-cert/tls.crt /ovn-ca/ca-bundle.crt && \
+                          ovn-sbctl -t 5 set-connection pssl:{{.OVN_SB_PORT}} -- set connection . inactivity_probe=0 )
+                do
```

Comment:
@squeed This hack only works at install time. Setting the listen port was done after the sdb was synced. Once set the port is opened immediately at sdb start, it doesn't wait for the resync to complete.

Reply:
Hmm, not sure what you mean by this. How should this line read, then?

Reply:
The whole `postStart` thing is really insane anyway... there must be a nicer way to do this

Reply:
@squeed @dcbw 
The install uses the unix socket to set up and sync the sdb. When it is ready, the ports are opened. On successive reboots of sdb the port is immediately opened and it starts serving requests. It doesn't wait for sync to complete (this is what prevented install from working when the ports were opened early). There needs to be a lock that prevents this.

ovnkube should set the ports when the DB is set up, it should not be in the script. At present there is a race since ovnkube is working in parallel with the script. This appears to work because on install (first boot) there is not much to. It shouldn't be done in the script.

The "while" is not needed since ovnkube has been working with the sdb over the unix socket. The first attempt will work.

Reply:
I see that it configures the endpoints in ovnkube-master, but where does it wait for northd to do a sync loop? And why does this matter? What happens if the southd is listening as configuration is enabled?

Reply:
sdb immediately serves requests when it starts. When I originally did the ovn bringup this caused pods to go into crashloopbackoff. I had to hold off sdb until it was in sync with the api server. So I used the unix socket to set it up and when that was done setup the network ports.

Reply:
The cluster knows what should be running. Bad things happen whenever the sdb has a different  world view than the cluster. sdb it not truth. It is a cache. It needs to be treated like that.

Reply:
@squeed @danwinship  ^^

Reply:
This is all moot, because ovn-kubernetes communicates over a unix socket to the local DB. So we can't use it's feature that programs the endpoints in to the nb/sbdb.

Reply:
Hmm. This is all moot, because ovnkube-master uses a unix socket to talk to the sbdb. It needs an independent way to learn its endpoints.

Reply:
@squeed It uses the unix socket to sync up the sdb. The sdb must be setup before the tcp ports are opened or pods go into crashloopbackoff.  This was done as a hack at install time. Restarts and reboots are still a problem since sdb remembers to immediately open the ports and start serving. 
At present we start master and then set up the tcp ports. The expectation is that on install, master will get the sdb setup before the ports get setup (this is a race). This does not work for restarts/reboots.
All this should be moved to ovnkube --init-master

Reply:
Why do pods start crashlooping?

Reply:
@squeed we have code in config.go to do the set-ssl operation already. This doesn't need to be done in the script.

Reply:
Well, unless you want to split the master up and run the sbdb on a different node than master... but this PR doesn't do that. I suppose we should have a conversation about how much the master should handle DB configuration; does the admin/container need to set up the authentication and ports, or ovnkube-master?

Doing it all in the containers would be great and we could delete code from Go. But this PR doesn't do that...

Reply:
@dcbw @squeed we need to use unix sockets while setting up sdb. When sdb listens on the network port it handles requests. This must be prevented while setting up sdb otherwise pods go into CrashLoopBackup. The listening ports need to be set up after DB sync completes.  (This was discovered in original OVN install work).

When sdb comes up it immediately listens on all configured ports and handles requests. This is  a problem whenever the sdb and apiserver are out of sync. The unix socket hack only works during install, not restart or reboot.

Reply:
@dcbw 

> we have code in config.go to do the set-ssl operation already.

Reading the code, I couldn't figure out how to get go to set-ssl and also use a unix socket.

Reply:
@pecameron 

>  This must be prevented while setting up sdb otherwise pods go into CrashLoopBackup.

As I already explained, I don't care if they go in CrashLoopBackoff, because they quickly leave it. Besides, this was fixed when we stopped getting the chassis id from the sbdb. Also, none of this will matter, since we have to open the sbdb port for raft.

Reply:
@pecameron we do have Numans HA pr which won't update the endpoints that the nodes use until the southbound is synced with northbound. So now the way this will work is that ovn-controller won't even know where to look for the southbound until things are already synced, which avoids the southbound port open issue.